### PR TITLE
Added script to build jquery chosen 1.0.0 package on rhel 8.4

### PR DESCRIPTION
**Pull request by sachinkakatkar**
## Checklist

- [ ] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [ ] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

File: c/chosen/chosen_rhel_8_4.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# -----------------------------------------------------------------------------
+#
+# Package       : chosen
+# Version       : 1.0.0
+# Source repo   : https://github.com/harvesthq/chosen/archive/1.0.0.zip
+# Tested on     : UBI 8
+# Script License: Apache License, Version 2 or later
+# Maintainer    : sachin.kakatkar@ibm.com
+#
```
Comment (kandarpamalipeddi) [inline c/chosen/chosen_rhel_8_4.sh:9]: From now on wards, please start adding Language to the Script header.
E.g:
```
# Maintainer : sachin.kakatkar@ibm.com
# Language : Node
```

File: c/chosen/chosen_rhel_8_4.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Disclaimer: This script has been tested in root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+#To Run: ./chosen_rhel_8_4.sh
+dnf install git wget npm make gcc-c++ bzip2 unzip ruby -y
+wget https://github.com/harvesthq/chosen/archive/1.0.0.zip
```
Comment (kandarpamalipeddi) [inline c/chosen/chosen_rhel_8_4.sh:19]: Never download the ZIP/tar files for build and validation.
Please use the `git clone` for the repo and `git checkout` for the Version tag/branch.

@seth-priya : Correct me if I am wrong on above comment.

Use package version into a variable and take it from command line parameter. If it is not provided, you can take one version (currently on which version we are working) as default version for build and validation.

Code should be as below :
```
PACKAGE_VERSION=${1:-1.0.0}
git clone https://github.com/harvesthq/chosen
cd chosen
git checkout $PACKAGE_VERSION
```

This is a generic comment and applicable for all the build-scripts.

File: c/chosen/chosen_rhel_8_4.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#
+# Disclaimer: This script has been tested in root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+#To Run: ./chosen_rhel_8_4.sh
+dnf install git wget npm make gcc-c++ bzip2 unzip ruby -y
```
Comment (kandarpamalipeddi) [inline c/chosen/chosen_rhel_8_4.sh:18]: Avoid installing unwanted/unused dependencies.
Do we need `make, gcc-c++` for building this package ?

**Review by kandarpamalipeddi — CHANGES_REQUESTED** [review]

File: c/chosen/chosen_rhel_8_4.sh (hunk 2 above)
Comment (sachinkakatkar) [reply c/chosen/chosen_rhel_8_4.sh:19]: 1.0.0 is not available in git hub repo.

File: c/chosen/chosen_rhel_8_4.sh (hunk 3 above)
Comment (sachinkakatkar) [reply c/chosen/chosen_rhel_8_4.sh:18]: Yes those packages are required.

File: c/chosen/chosen_rhel_8_4.sh (hunk 1 above)
Comment (sachinkakatkar) [reply c/chosen/chosen_rhel_8_4.sh:9]: Okay

File: c/chosen/chosen_rhel_8_4.sh (hunk 2 above)
Comment (kandarpamalipeddi) [reply c/chosen/chosen_rhel_8_4.sh:19]: I can see version is available at : https://github.com/harvesthq/chosen/tree/1.0.0

I tried checkout myself and it is working fine.
```
~]# git clone https://github.com/harvesthq/chosen; cd chosen
Cloning into 'chosen'...
remote: Enumerating objects: 7464, done.
remote: Total 7464 (delta 0), reused 0 (delta 0), pack-reused 7464
Receiving objects: 100% (7464/7464), 2.98 MiB | 23.32 MiB/s, done.
Resolving deltas: 100% (4595/4595), done.

 chosen]# git checkout 1.0.0
Note: switching to '1.0.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at b34a5d8 Updating to 1.0
```

image

File: c/chosen/chosen_rhel_8_4.sh (hunk 3 above)
Comment (kandarpamalipeddi) [reply c/chosen/chosen_rhel_8_4.sh:18]: If you use the git hub repo (Github repo details can be found in below comment), you don't need `bzip2 unzip`.

I don't see any make command in the script .
I don't see any dependency on gcc and c++.

You only required, `git npm ruby`.

File: c/chosen/chosen_rhel_8_4.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+unzip 1.0.0.zip
+mv chosen-1.0.0 chosen
+cd chosen
+npm install && gem install bundler && bundle install
+npm audit fix
+npm audit fix --force
+npm install -g grunt-cli
+npm install -g grunt
+npm install --dev coffeescript
+npm install && gem install bundler && bundle install
```
Comment (kandarpamalipeddi) [inline c/chosen/chosen_rhel_8_4.sh:29]: Do we need this again ?

once we installed the bundler, do we need it install again?

**Review by kandarpamalipeddi — CHANGES_REQUESTED** [review]

File: c/chosen/chosen_rhel_8_4.sh [hunk 5]
```diff
@@ -2,8 +2,9 @@
 #
 # Package       : chosen
 # Version       : 1.0.0
-# Source repo   : https://github.com/harvesthq/chosen/archive/1.0.0.zip
+# Source repo   : https://github.com/harvesthq/chosen
 # Tested on     : UBI 8
+# Language      : HTML
```
Comment (kandarpamalipeddi) [inline c/chosen/chosen_rhel_8_4.sh:7]: Use Node as Language for NPMs.

Please check the templates at : https://github.com/ppc64le/build-scripts/blob/master/templates/ for more details.

File: c/chosen/chosen_rhel_8_4.sh [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Disclaimer: This script has been tested in root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+#To Run: ./chosen_rhel_8_4.sh
+dnf install git make gcc-c++ ruby -y
```
Comment (kandarpamalipeddi) [inline c/chosen/chosen_rhel_8_4.sh:19]: Do we still need make/gcc-c++ ?

**Review by kandarpamalipeddi — CHANGES_REQUESTED** [review]
Can you please paste the final results ?
Logs of the `npm` commands ?

File: c/chosen/chosen_rhel_8_4.sh (hunk 6 above)
Comment (sachinkakatkar) [reply c/chosen/chosen_rhel_8_4.sh:19]: Yes they are required

File: c/chosen/chosen_rhel_8_4.sh (hunk 5 above)
Comment (sachinkakatkar) [reply c/chosen/chosen_rhel_8_4.sh:7]: Okay

File: c/chosen/chosen_rhel_8_4.sh (hunk 4 above)
Comment (sachinkakatkar) [reply c/chosen/chosen_rhel_8_4.sh]: Okay removed

**Comment by sachinkakatkar** [comment]
npm WARN grunt-build-gh-pages@0.0.4 requires a peer of grunt@~0.4.1 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-clean@0.4.1 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-coffee@0.6.7 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-compass@0.3.0 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-concat@0.1.3 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-copy@0.4.1 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-jshint@0.1.1 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-nodeunit@0.1.2 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-shell@0.2.2 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN chosen@1.0.0 No license field.

audited 627 packages in 3.5s

47 packages are looking for funding
  run `npm fund` for details

found 47 vulnerabilities (6 low, 11 moderate, 23 high, 7 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
Successfully installed bundler-2.3.4
1 gem installed
Your RubyGems version (2.7.6.3)) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for all
non-root users on this machine.
Using bundler 2.3.4
Using chunky_png 1.2.8
Using fssm 0.2.10
Using sass 3.2.9
Using compass 0.12.2
Bundle complete! 1 Gemfile dependency, 5 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.

**Comment by sachinkakatkar** [comment]
This PR closing because chosen package is deprecated and it is failing with normal installation and install success with force installation.

**Pull request closed without merging**